### DISCUSSION TITLE
[Question]  How could I display a documment's array as a collection?

### DISCUSSION THREAD

**Discussion by Evaldash** [firecmsco/firecms#245] [Q&A]
I have a string array in one document, and would like to display it in a page, so it could be edited like a collection. How could I do that?

**Comment by fgatti675** [comment]
Hi @Evaldash, that is currently not straightforward in the current version.
FireCMS exposes many of its internal components thought: 
- You could create your own custom component that uses `useDataSource` for fetching and saving data.
- For the table there is a `Table` component (not documented) that is used for rendering the table, but does not provide the inline editing capabilities.
The component that allows inline editing is called `CollectionTable` but it provides much more functionality than what you need, since it is bound to a collection and it won't help you.
If you still want to go down the `Table` and `useDataSource` road, you can build the editable cells copying this file:
https://github.com/Camberi/firecms/blob/master/src/core/components/CollectionTable/internal/fields/TableInput.tsx
since it is not exposed by the library.
Let me know how it goes ;)
